test/e2e/conformance_test.go

```diff
@@ -0,0 +1,201 @@
+			machineDeploymentName := clusterName + "-worker"
+			createMachineDeployment(namespace, clusterName, machineDeploymentName, 2)
+
+			By("Running conformance on the workload cluster")
+			err = runConformance(testTmpDir, namespace, clusterName)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Deleting the Cluster")
+			deleteCluster(namespace, clusterName)
+		})
+	})
+})
+
+type sonobuoyConfig struct {
+	SonobuoyVersion string
+	K8sVersion      string
+}
+
+func runConformance(tmpDir, namespace, clusterName string) error {
+	cluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      clusterName,
+		},
+	}
+	kubeConfigData, err := kubeconfig.FromSecret(kindClient, cluster)
+	if err != nil {
+		return errors.Wrap(err, "couldn't get kubeconfig of workload cluster")
+	}
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(kubeConfigData)
+	if err != nil {
+		return errors.Wrap(err, "couldn't get rest config from kubeconfig")
+	}
+
+	sonobuoyKubeCli, err := sonodynamic.NewAPIHelperFromRESTConfig(restConfig)
+	if err != nil {
+		return err
+		return errors.Wrap(err, "couldn't get rest config from kubeconfig")
+	}
+
+	sonobuoyClient, err := client.NewSonobuoyClient(restConfig, sonobuoyKubeCli)
+	if err != nil {
+		return errors.Wrap(err, "couldn't create sonobuoy client")
+	}
+
+	k8sClient, err := sonobuoyClient.Client()
+	if err != nil {
+		return errors.Wrap(err, "couldn't retrieve k8s client")
+	}
+
+	apiVersion, err := k8sClient.Discovery().ServerVersion()
+	if err != nil {
+		return errors.Wrap(err, "couldn't get workload cluster's k8s version")
+	}
+
+	tmpl, err := template.ParseFiles("sonobuoy-config-tmpl.yaml")
+	if err != nil {
+		return errors.Wrap(err, "couldn't parse sonobuoy config template")
+	}
+
+	sonobuoyConfigPath := path.Join(tmpDir, clusterName+"-sonobuoy-config.yaml")
+	fileP, err := os.Create(sonobuoyConfigPath)
+	if err != nil {
+		return errors.Wrap(err, "couldn't create sonobuoy config file")
+	}
+
+	sbConfig := sonobuoyConfig{SonobuoyVersion: "v0.16.2", K8sVersion: apiVersion.GitVersion}
+	err = tmpl.Execute(fileP, sbConfig)
+	if err != nil {
+		return errors.Wrap(err, "couldn't execute template")
+	}
+
+	runConfig := &client.RunConfig{
+		Wait:    time.Duration(4) * time.Hour,
+		GenFile: sonobuoyConfigPath,
+	}
+	if err := sonobuoyClient.Run(runConfig); err != nil {
+		return errors.Wrap(err, "error attempting to run sonobuoy")
+	}
+	reader, ec, err := sonobuoyClient.RetrieveResults(&client.RetrieveConfig{Namespace: "sonobuoy"})
+	if err != nil {
+		return errors.Wrap(err, "couldn't retrieve sonobuoy results")
+	}
+	eg := &errgroup.Group{}
+	eg.Go(func() error { return <-ec })
+	eg.Go(func() error {
+		filesCreated, err := client.UntarAll(reader, ".", "")
+		if err != nil {
+			return errors.Wrap(err, "couldn't untar sonobuoy results")
+		}
+		for _, name := range filesCreated {
+			fmt.Fprintf(GinkgoWriter, "Find conformance test results at: test/e2e/%s\n", name)
```

Comment:
Unless I'm misreading this, I believe that the `filesCreated` is the tarball itself and so the path "test/e2e/<name>" isn't right. 

E.g. when you run `sonobuoy retrieve` it prints the filenames of the tarball it downloaded. Maybe this was misleading because of the `UntarAll` but it is untarring the response from the server which in most cases results in a single tarball name in `filesCreated` but (in theory) could be more tarballs.